its/src/test/java/com/sonar/it/scanner/msbuild/utils/ContextExtension.java

```diff
@@ -77,4 +81,36 @@ private static <T> T ensureNotNull(ThreadLocal<T> threadLocal) {
       return value;
     }
   }
+
+  private void checkWorkloadPrerequisites(ExtensionContext context) {
+    Optional<WorkloadPrerequisite> workloadPrereq = Optional
+      .ofNullable(context.getRequiredTestMethod().getAnnotation(WorkloadPrerequisite.class))
+      .or(() -> Optional.ofNullable(context.getRequiredTestClass().getAnnotation(WorkloadPrerequisite.class)));
+
+    workloadPrereq.ifPresent(prerequisite -> {
+      // Usually workloads are a VisualStudio thing, so we want to ensure that the test is run on Windows
+      // If needed, this can be extended to other OSes in the future
+      verifyWindowsOnly(context);
+      if (!prerequisite.value().isInstalled()) {
+        throw new IllegalStateException("Required workload not installed: " + prerequisite.value().getId());
+      }
+    });
+  }
+
+  private void verifyWindowsOnly(ExtensionContext context) {
+    Optional<EnabledOnOs> enabledOnOs = Optional
+      .ofNullable(context.getRequiredTestMethod().getAnnotation(EnabledOnOs.class))
+      .or(() -> Optional.ofNullable(context.getRequiredTestClass().getAnnotation(EnabledOnOs.class)));
+
+    if (enabledOnOs.isEmpty()) {
+      throw new IllegalStateException("WorkloadPrerequisite annotation requires tests to be Windows-only, " +
+        "add @EnabledOnOs(OS.WINDOWS) to the test class or method.");
```

Comment:
To leave both options open. We won't remember the details few years from now
```suggestion
        "add @EnabledOnOs(OS.WINDOWS) to the test class or method. Or extend WorkloadPrerequisite logic.");
```